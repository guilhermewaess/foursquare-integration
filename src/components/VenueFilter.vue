<template>
  <v-container class="pa-0">
    <form @submit.prevent="search()">
      <v-layout row>
        <v-flex xs12
                sm2>
          <v-select v-model="filter.section"
                    :items="sections"
                    item-text="name"
                    item-value="value"
                    label="Section"
                    class="pr-4">
          </v-select>
        </v-flex>
        <v-flex xs12
                sm5>
          <v-text-field v-model="filter.search"
                        label="Search Term"
                        prepend-icon="search">
          </v-text-field>
        </v-flex>
        <v-flex xs12
                sm3>
          <v-switch label="Open Now"
                    v-model="filter.openNow"
                    color="primary"
                    class="pl-5">
          </v-switch>
        </v-flex>
        <v-flex xs12
                sm2>
          <v-btn color="primary"
                 type="submit"
                 block
                 outline
                 class="mt-3"
                 dark>Search</v-btn>
        </v-flex>
      </v-layout>
    </form>
  </v-container>
</template>

<script>
export default {
  name: 'VenueFilter',
  data() {
    return {
      filter: {
        search: '',
        section: '',
        openNow: true,
      },
      sections: [
        { name: 'All', value: '' },
        { name: 'Food', value: 'food' },
        { name: 'Drinks', value: 'drinks' },
        { name: 'Coffee', value: 'coffee' },
        { name: 'Shops', value: 'shops' },
        { name: 'Arts', value: 'arts' },
        { name: 'Outdoors', value: 'outdoors' },
        { name: 'Sights', value: 'sights' },
        { name: 'Trending', value: 'trending' },
        { name: 'NextVenues', value: 'nextVenues' },
        { name: 'TopPicks', value: 'topPick' },
      ],
    };
  },
  methods: {
    search() {
      console.log(this.filter);
    },
  },
};
</script>
