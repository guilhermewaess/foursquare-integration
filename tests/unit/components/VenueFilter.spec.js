import { shallowMount } from '@vue/test-utils';
import VenueFilter from '@/components/VenueFilter.vue';

describe('VenueFilter', () => {
  let component;
  beforeEach(() => {
    component = shallowMount(VenueFilter, {
      stubs: ['v-container', 'v-layout', 'v-flex', 'v-select', 'v-switch', 'v-btn', 'v-text-field'],
    });
  });
  it('should have name', () => {
    expect(component.name()).toEqual('VenueFilter');
  });
  describe('snapshot', () => {
    it('should render', () => {
      expect(component.element).toMatchSnapshot();
    });
  });
  describe('data', () => {
    it('should start filter properties', () => {
      expect(component.vm.filter).toEqual({
        query: '',
        section: '',
        openNow: true,
      });
    });
    it('should start sections options', () => {
      expect(component.vm.sections).toEqual([
        { name: 'All', value: '' },
        { name: 'Food', value: 'food' },
        { name: 'Drinks', value: 'drinks' },
        { name: 'Coffee', value: 'coffee' },
        { name: 'Shops', value: 'shops' },
        { name: 'Arts', value: 'arts' },
        { name: 'Outdoors', value: 'outdoors' },
        { name: 'Sights', value: 'sights' },
        { name: 'Trending', value: 'trending' },
        { name: 'NextVenues', value: 'nextVenues' },
        { name: 'TopPicks', value: 'topPick' },
      ]);
    });
  });
  describe('methods', () => {
    it('search should emit onFilter event with filter', () => {
      component.vm.search();
      const emittedPayload = component.emitted().onFilter[0][0];
      expect(emittedPayload).toEqual(component.vm.filter);
    });
  });
});
