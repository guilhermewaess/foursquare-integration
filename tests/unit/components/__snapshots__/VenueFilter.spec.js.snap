// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenueFilter snapshot should render 1`] = `
<v-container-stub
  class="pa-0"
>
  <form>
    <v-layout-stub
      row=""
    >
      <v-flex-stub
        sm2=""
        xs12=""
      >
        <v-select-stub
          class="pr-4"
          item-text="name"
          item-value="value"
          items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
          label="Section"
        />
      </v-flex-stub>
       
      <v-flex-stub
        sm5=""
        xs12=""
      >
        <v-text-field-stub
          label="Search Term"
          prepend-icon="search"
        />
      </v-flex-stub>
       
      <v-flex-stub
        sm3=""
        xs12=""
      >
        <v-switch-stub
          class="pl-5"
          color="primary"
          label="Open Now"
        />
      </v-flex-stub>
       
      <v-flex-stub
        sm2=""
        xs12=""
      >
        <v-btn-stub
          block=""
          class="mt-3"
          color="primary"
          dark=""
          outline=""
          type="submit"
        >
          Search
        </v-btn-stub>
      </v-flex-stub>
    </v-layout-stub>
  </form>
</v-container-stub>
`;
